bam_to_mate_hist.py

```diff
@@ -11,147 +11,291 @@
-
-		fig2 = plt.figure()
-		plt.hist(dists, bins=xrange(0,20000, 500))
-		ax = fig2.add_subplot(111)
-		ax.set_xlim(0, 20000)
-		ax.set_ylim(0.5, num_reads * 2)
-		plt.yscale("log", nonposy="clip")
-		plt.title("Mate distance distribution for sample\n" + bamfile_handle+"\nfor first " + str(num_reads)+ " reads")
-		plt.xlabel("Distance between read pair mates in Hi-C mapping")
-		plt.ylabel("Number of reads")
-		pdf.savefig(fig2)
-		plt.close()
+	'''
+    num_reads = len(dists)
+    # with PdfPages(outfile_name) as pdf:
+    fig1 = plt.figure()
+    plt.hist(dists, bins=40)
+    ax = fig1.add_subplot(111)
+    ax.set_ylim(0.5, num_reads * 2)
+    plt.yscale("log", nonposy="clip")
+    plt.title("\nMate distance distribution for first " + str(num_reads) + " reads for sample\n" + bamfile_handle)
+    plt.xlabel("Distance between read pair mates in Hi-C mapping")
+    plt.ylabel("Number of reads")
+    fig1.savefig(outfile_name + "_long.png")
+    plt.close(fig1)
+
+    fig2 = plt.figure()
+    plt.hist(dists, bins=xrange(0, 20000, 500))
+    ax = fig2.add_subplot(111)
+    ax.set_xlim(0, 20000)
+    ax.set_ylim(0.5, num_reads * 2)
+    plt.yscale("log", nonposy="clip")
+    plt.title("Mate distance distribution for first " + str(num_reads) + " reads for sample\n" + bamfile_handle)
+    plt.xlabel("Distance between read pair mates in Hi-C mapping")
+    plt.ylabel("Number of reads")
+    fig2.savefig(outfile_name + "_short.png")
+    plt.close(fig2)
+
+
+def make_pdf_report(qc_repo_path, stat_dict, outfile_name):
+    '''Make the pdf report using the template, the
+
+    Args:
+        qc_repo_path (str): path the qc tool repo, which holds markdown and style templates
+        stat_dict ({str: str/float}): a dictionary of relevant data (stats, file paths) regarding the lib to be used in the report.
+        outfile_name (str): path to which the report shall be written.
+
+    '''
+    options = {
+        'page-size': 'A4',
+        'dpi': 350,
+        'margin-top': '0.75in',
+        'margin-right': '0.75in',
+        'margin-bottom': '0.75in',
+        'margin-left': '0.75in',
+        'encoding': "UTF-8",
+        'custom-header': [
+            ('Accept-Encoding', 'gzip')
+        ],
+        'no-outline': None
+    }
+
+    template_path = os.path.join(qc_repo_path, "collateral", "HiC_QC_report_template.md")
+    style_path = os.path.join(qc_repo_path, "collateral", "style.css")
+    if not os.path.exists(template_path):
+        UserWarning("can't find markdown template at {0}! skipping making template.".format(
+            qc_repo_path)
+        )
+        sys.exit()
+
+    with open(template_path) as file:
+        str = file.read()
+        sub_str = str.format(**stat_dict)  # splat the statistics and path into the markdown, render as html
+        # print sub_str
+        html = md.markdown(sub_str, extensions=['tables', 'nl2br'])
+
+        # write out just html
+        html_out = open(outfile_name + "_qc_report.html", 'w')
```

Comment:
My usual switch to context manager comment.

Reply:
done